android/src/main/java/com/swmansion/rnscreens/ScreenStackFragment.kt

```diff
@@ -154,7 +154,12 @@ class ScreenStackFragment :
     }
 
     override fun onPrepareOptionsMenu(menu: Menu) {
-        updateToolbarMenu(menu)
+        // If the screen is a transparent modal with hidden header we don't want to update the toolbar
+        // menu because it may erase the menu of the previous screen (which is still visible in these
+        // circumstances). See here: https://github.com/software-mansion/react-native-screens/issues/2271
+        if (screen.stackPresentation != Screen.StackPresentation.TRANSPARENT_MODAL || screen.headerConfig?.isHeaderHidden == false) {
+            updateToolbarMenu(menu)
+        }
```

Comment:
There is a private `isTransparent` method in `ScreenStack.Companion`. I would like you to move this method to the `Screen`, make it public and use it here.

I want you to do this, because we will have more transparent presentations introduced with #2045 and I want this logic to stay working.